qi-lib/macro.rkt

```diff
@@ -108,3 +111,40 @@
          (define-dsl-syntax form-name qi-macro
            (make-qi-foreign-syntax-transformer #'form-name))
          ...)]))
+
+(begin-for-syntax
+  (define (op-transformer info spec)
```

Comment:
Surrounded by `define-syntax`es, I worry that it would be visually confusing as it could be easily mistaken for another `define-syntax`. It seems safer to leave as is.